Sources_v3/ChatClient.swift

```diff
@@ -52,6 +52,19 @@ public class Client<ExtraData: ExtraDataTypes> {
     /// observing the current state of the system and perform work if needed (i.e. when a new message pending sent appears in the
     /// database, a worker tries to send it.)
     private(set) var backgroundWorkers: [Worker]!
+            
+    /// The notification center used to send and receive notifications about incoming events.
+    private(set) lazy var eventNotificationCenter: EventNotificationCenter = {
+        // Create middlewares.
+        let eventMiddlewares: [EventMiddleware] = [
+            EventDataProcessorMiddleware<ExtraData>(database: databaseContainer)
+        ]
+        
+        // Create notification center
+        let notificationCenter = environment.createNotificationCenter()
```

Comment:
This should probably be inside the `EventNotificationCenter` and its own environment

Reply:
made a subclass in 9ccf3e7d5bc8ffbd0021704e53abe9deed9320bb